### ISSUE TITLE
3.0.6 upgrade to 3.1.2, run error: ER_TOO_LONG_KEY: Specified key was too long; max key length is 767 bytes

### ISSUE THREAD

**Issue by xiaoer3301** [strapi/strapi#7273]
**Describe the bug**
Running `NODE_ENV = production npm run start` reported the following error.

**Steps to reproduce the behavior**

1. Modify package.json strapi related version to 3.1.2
2. server.js
``` js
admin: {
      auth: {
        secret: env('ADMIN_JWT_SECRET', 'raQK1xx0PgsRqzLBFd7pqXVfCXhavsOretSJOel3DgQh0cFMt4TwdR968lvq0RNVqvWpppNYvlZhIJdH4YHfwA=='),
      },
    },
```
3. npm i
4. NODE_ENV=production npm run build -- --clean
5. NODE_ENV=production npm run start

**Code snippets**
```
[2020-07-31T03:40:01.437Z] error Migration failed
[2020-07-31T03:40:01.438Z] error Error: ER_TOO_LONG_KEY: Specified key was too long; max key length is 767 bytes
    at Query.Sequence._packetToError (/home/strapi/node_modules/mysql/lib/protocol/sequences/Sequence.js:47:14)
    at Query.ErrorPacket (/home/strapi/node_modules/mysql/lib/protocol/sequences/Query.js:79:18)
    at Protocol._parsePacket (/home/strapi/node_modules/mysql/lib/protocol/Protocol.js:291:23)
    at Parser._parsePacket (/home/strapi/node_modules/mysql/lib/protocol/Parser.js:433:10)
    at Parser.write (/home/strapi/node_modules/mysql/lib/protocol/Parser.js:43:10)
    at Protocol.write (/home/strapi/node_modules/mysql/lib/protocol/Protocol.js:38:16)
    at Socket.<anonymous> (/home/strapi/node_modules/mysql/lib/Connection.js:88:28)
    at Socket.<anonymous> (/home/strapi/node_modules/mysql/lib/Connection.js:526:10)
    at Socket.emit (events.js:315:20)
    at addChunk (_stream_readable.js:295:12)
    at readableAddChunk (_stream_readable.js:271:9)
    at Socket.Readable.push (_stream_readable.js:212:10)
    at TCP.onStreamRead (internal/stream_base_commons.js:186:23)
    --------------------
    at Protocol._enqueue (/home/strapi/node_modules/mysql/lib/protocol/Protocol.js:144:48)
    at Connection.query (/home/strapi/node_modules/mysql/lib/Connection.js:198:25)
    at /home/strapi/node_modules/knex/lib/dialects/mysql/index.js:130:18
    at Promise._execute (/home/strapi/node_modules/bluebird/js/release/debuggability.js:384:9)
    at Promise._resolveFromExecutor (/home/strapi/node_modules/bluebird/js/release/promise.js:518:18)
    at new Promise (/home/strapi/node_modules/bluebird/js/release/promise.js:103:10)
    at Client_MySQL._query (/home/strapi/node_modules/knex/lib/dialects/mysql/index.js:124:12)
    at Client_MySQL.query (/home/strapi/node_modules/knex/lib/client.js:158:17)
    at /home/strapi/node_modules/knex/lib/transaction.js:331:24
    at Promise._execute (/home/strapi/node_modules/bluebird/js/release/debuggability.js:384:9)
    at Promise._resolveFromExecutor (/home/strapi/node_modules/bluebird/js/release/promise.js:518:18)
    at new Promise (/home/strapi/node_modules/bluebird/js/release/promise.js:103:10)
    at Client_MySQL.trxClient.query (/home/strapi/node_modules/knex/lib/transaction.js:326:12)
    at Runner.query (/home/strapi/node_modules/knex/lib/runner.js:135:36)
    at Runner.<anonymous> (/home/strapi/node_modules/knex/lib/runner.js:228:25)
    at Runner.tryCatcher (/home/strapi/node_modules/bluebird/js/release/util.js:16:23)
```

**System**

-Node.js version: v12.18.3
-NPM version:6.14.6
-Strapi version: 3.1.2
-Database: 10.1.37-MariaDB, libmysql - mysqlnd 5.0.12-dev - 20150407
-Operating system: centos

**Comment by derrickmehaffy** [comment]
one of your models must have a very long name 🤔 are you using a lot of components?

**Comment by xiaoer3301** [comment]
> one of your models must have a very long name 🤔 are you using a lot of components?
Please look at this error.
```
SQL 查询：


--
-- 表的索引 `strapi_administrator`
--
ALTER TABLE `strapi_administrator`
  ADD PRIMARY KEY (`id`),
  ADD UNIQUE KEY `strapi_administrator_email_unique` (`email`)
MySQL 返回： 文档

#1071 - Specified key was too long; max key length is 767 bytes
```

**Comment by xiaoer3301** [comment]
After I upgraded the mysql version, this problem was solved. The available versions are: 10.4.13-MariaDB, libmysql-mysqlnd 7.4.8.
The upgrade process was painful and almost lost the database.

**Comment by geekplux** [comment]
This error:
 
> ER_TOO_LONG_KEY: Specified key was too long; max key length is 767 bytes

would cause login failed even when first time creating superuser - same 401 as https://github.com/strapi/strapi/issues/7140
